day7/part2.go

```diff
@@ -37,7 +37,7 @@ fuel possible so they can make you an escape route! How much fuel must they
 spend to align to that position?
 */
 func Part2(path string) (int, error) {
-	return Part2Opts(path, Naive, ForLoop) // fastest approach on my machine
+	return Part2Opts(path, NoCache, WaitGroup) // fastest approach on my machine
```

Comment:
new benchmarks

```text
goos: linux
goarch: amd64
pkg: ryepup/advent2021/day7
cpu: Intel(R) Core(TM) i7-10875H CPU @ 2.30GHz
BenchmarkPart2/P:0_C:0-16         	     266	   4414719 ns/op	   49288 B/op	       9 allocs/op
BenchmarkPart2/P:0_C:1-16         	      20	  57884142 ns/op	  223020 B/op	     114 allocs/op
BenchmarkPart2/P:0_C:2-16         	      22	  52782145 ns/op	  222749 B/op	     112 allocs/op
BenchmarkPart2/P:0_C:3-16         	      27	  42251200 ns/op	  222900 B/op	     112 allocs/op
BenchmarkPart2/P:1_C:0-16         	    1386	    828358 ns/op	  245490 B/op	    3782 allocs/op
BenchmarkPart2/P:1_C:1-16         	       9	 153201855 ns/op	  449887 B/op	    4208 allocs/op
BenchmarkPart2/P:1_C:2-16         	      21	  76002822 ns/op	  439538 B/op	    4095 allocs/op
--- FAIL: BenchmarkPart2/P:1_C:3
    /workspaces/advent-of-code-2021/day7/day_test.go:47: invalid strategy combination
```

Now the "expensive" part moved from `rawFuelCost` to iterating through the nested for loop for `targets` and `positions` calculating the cost of each possible solution.

Winner by an order of magnitude: waitgroups w/ no cost cache. This solution got 100x faster from the last benchmark.

Honorable mention to for loop w/ no cost cache, which also gained 100x over last time.